core/src/main/java/org/web3j/protocol/websocket/WebSocketClient.java

```diff
@@ -0,0 +1,60 @@
+package org.web3j.protocol.websocket;
+
+import java.net.URI;
+
+import org.java_websocket.handshake.ServerHandshake;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Web socket client implementation that connects to a specify URI. Allows to provide a listener
+ * that will be called when a new message is received by the client.
+ */
+public class WebSocketClient extends org.java_websocket.client.WebSocketClient {
+
+    private static final Logger log = LoggerFactory.getLogger(WebSocketClient.class);
+
+    private WebSocketListener listener;
+
+    public WebSocketClient(URI serverUri) {
+        super(serverUri);
+    }
+
+    @Override
+    public void onOpen(ServerHandshake serverHandshake) {
+        log.info("Opened WebSocket connection to {}", uri);
+    }
+
+    @Override
+    public void onMessage(String s) {
+        try {
+            log.debug("Received message {} from server {}", s, uri);
+            listener.onMessage(s);
+            log.debug("Processed message {} from server {}", s, uri);
```

Comment:
I don't think we need both debug statements

Reply:
Removed second message.